diff --git a/drivers/thermal/virtual_sensor_cpufreq_cooling.c b/drivers/thermal/virtual_sensor_cpufreq_cooling.c
index 99c4cae1..705993b1 100644
--- a/drivers/thermal/virtual_sensor_cpufreq_cooling.c
+++ b/drivers/thermal/virtual_sensor_cpufreq_cooling.c
@@ -29,7 +29,7 @@
 #include <linux/thermal.h>
 #include <linux/cpumask.h>
 #include <linux/cpu_cooling.h>
-#include <thermal_core.h>
+#include "thermal_core.h"
 
 #define MAX_STATE	4 /* 0-4 total 5 */
 
