diff --git a/drivers/sharp/shmds/mpu9150/inv_gyro.h b/drivers/sharp/shmds/mpu9150/inv_gyro.h
index 1d920c27..99686f81 100755
--- a/drivers/sharp/shmds/mpu9150/inv_gyro.h
+++ b/drivers/sharp/shmds/mpu9150/inv_gyro.h
@@ -705,7 +705,6 @@ int create_device_attributes(struct device *dev,
 void remove_device_attributes(struct device *dev,
 	struct device_attribute **attrs);
 int set_3050_bypass(struct inv_gyro_state_s *st, int enable);
-inline s64 get_time_ns(void);
 int inv_mpu3050_create_sysfs(struct inv_gyro_state_s *st);
 int inv_mpu3050_remove_sysfs(struct inv_gyro_state_s *st);
 int inv_get_accl_bias(struct inv_gyro_state_s *st, int *accl_bias_regular);
diff --git a/drivers/sharp/shmds/mpu9150/inv_mpu3050.c b/drivers/sharp/shmds/mpu9150/inv_mpu3050.c
index fade48de..c9e044e4 100755
--- a/drivers/sharp/shmds/mpu9150/inv_mpu3050.c
+++ b/drivers/sharp/shmds/mpu9150/inv_mpu3050.c
@@ -43,6 +43,8 @@
 #define MPU3050_ONE_MPU_TIME    (20)
 #define MPU3050_BOGUS_ADDR      (0x7F)
 
+extern s64 get_time_ns(void);
+
 int set_3050_bypass(struct inv_gyro_state_s *st, int enable)
 {
 	struct inv_reg_map_s *reg;
