diff --git a/drivers/video/samsung_extdisp/s3cfb_extdsp.h b/drivers/video/samsung_extdisp/s3cfb_extdsp.h
index 99f134c..11f4f37 100644
--- a/drivers/video/samsung_extdisp/s3cfb_extdsp.h
+++ b/drivers/video/samsung_extdisp/s3cfb_extdsp.h
@@ -151,7 +151,7 @@ struct s3cfb_extdsp_user_window {
 #define S3CFB_EXTDSP_GET_FREE_BUFFER		_IOW('F', 329, unsigned int)
 
 extern struct fb_ops			s3cfb_extdsp_ops;
-extern inline struct s3cfb_extdsp_global	*get_extdsp_global(int id);
+extern struct s3cfb_extdsp_global	*get_extdsp_global(int id);
 
 /* S3CFB_EXTDSP */
 extern int s3cfb_extdsp_enable_window(struct s3cfb_extdsp_global *fbdev, int id);
diff --git a/drivers/video/samsung_extdisp/s3cfb_extdsp_main.c b/drivers/video/samsung_extdisp/s3cfb_extdsp_main.c
index 4f9f2b2..dacd94d 100644
--- a/drivers/video/samsung_extdisp/s3cfb_extdsp_main.c
+++ b/drivers/video/samsung_extdisp/s3cfb_extdsp_main.c
@@ -43,7 +43,7 @@
 
 struct s3cfb_extdsp_extdsp_desc		*fbextdsp;
 
-inline struct s3cfb_extdsp_global *get_extdsp_global(int id)
+struct s3cfb_extdsp_global *get_extdsp_global(int id)
 {
 	struct s3cfb_extdsp_global *fbdev;
 
