Remove f_hidmodule implementation which causes kernel build to fail.
diff --git a/drivers/usb/gadget/function/f_hid.c b/drivers/usb/gadget/function/f_hid.c
index fef3c11d..f4e36021 100644
--- a/drivers/usb/gadget/function/f_hid.c
+++ b/drivers/usb/gadget/function/f_hid.c
@@ -1145,8 +1145,8 @@ static void __exit afunc_exit(void)
 	usb_function_unregister(&hidusb_func);
 }
 
-module_init(afunc_init);
-module_exit(afunc_exit);
+// module_init(afunc_init);
+// module_exit(afunc_exit);
 
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("Fabien Chouteau");
