diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 898eccb489c..aaa11f83a75 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -62,7 +62,6 @@
 #include "f_mtp.c"
 #include "f_accessory.c"
 #include "f_rndis.c"
-#include "rndis.c"
 #include "f_qc_ecm.c"
 #include "f_mbim.c"
 #include "f_qc_rndis.c"
diff --git a/drivers/usb/gadget/f_qc_rndis.c b/drivers/usb/gadget/f_qc_rndis.c
index 29022dcc822..4ed65a631fe 100644
--- a/drivers/usb/gadget/f_qc_rndis.c
+++ b/drivers/usb/gadget/f_qc_rndis.c
@@ -36,6 +36,11 @@
 #include "u_bam_data.h"
 #include <linux/rndis_ipa.h>
 
+
+#include <linux/usb_bam.h>
+
+#define LOG2_STATUS_INTERVAL_MSEC 5
+
 /*
  * This function is an RNDIS Ethernet port -- a Microsoft protocol that's
  * been promoted instead of the standard CDC Ethernet.  The published RNDIS
diff --git a/drivers/usb/gadget/f_rndis.c b/drivers/usb/gadget/f_rndis.c
index 74d261ab236..4d5337ee793 100644
--- a/drivers/usb/gadget/f_rndis.c
+++ b/drivers/usb/gadget/f_rndis.c
@@ -67,18 +67,10 @@
  */
 
 static unsigned int rndis_dl_max_pkt_per_xfer = 3;
-module_param(rndis_dl_max_pkt_per_xfer, uint, S_IRUGO | S_IWUSR);
-MODULE_PARM_DESC(rndis_dl_max_pkt_per_xfer,
-	"Maximum packets per transfer for DL aggregation");
 
 static unsigned int rndis_ul_max_pkt_per_xfer = 3;
-module_param(rndis_ul_max_pkt_per_xfer, uint, S_IRUGO | S_IWUSR);
-MODULE_PARM_DESC(rndis_ul_max_pkt_per_xfer,
-	"Maximum packets per transfer for UL aggregation");
 
 static unsigned int rx_trigger_enabled;
-module_param(rx_trigger_enabled, uint, S_IRUGO | S_IWUSR);
-MODULE_PARM_DESC(rx_trigger_enabled, "rx trigger_enable");
 
 struct f_rndis {
 	struct gether			port;
diff --git a/drivers/usb/gadget/rndis.c b/drivers/usb/gadget/rndis.c
index ca123b2e3e2..82636cfb999 100644
--- a/drivers/usb/gadget/rndis.c
+++ b/drivers/usb/gadget/rndis.c
@@ -32,6 +32,8 @@
 #include <linux/seq_file.h>
 #include <linux/netdevice.h>
 
+#include "f_qc_rndis.c"
+
 #include <asm/io.h>
 #include <asm/byteorder.h>
 #include <asm/unaligned.h>
diff --git a/drivers/usb/gadget/u_ether.c b/drivers/usb/gadget/u_ether.c
index ad51c79894c..052fd01d2bd 100644
--- a/drivers/usb/gadget/u_ether.c
+++ b/drivers/usb/gadget/u_ether.c
@@ -28,7 +28,7 @@
 #include <linux/notifier.h>
 #include <linux/cpufreq.h>
 #include "u_ether.h"
-
+#include "rndis.h"
 
 /*
