diff --git a/drivers/scsi/sg.c b/drivers/scsi/sg.c
index 15b09dee75d4..259ec2b5d31b 100644
--- a/drivers/scsi/sg.c
+++ b/drivers/scsi/sg.c
@@ -989,8 +989,6 @@ sg_ioctl(struct file *filp, unsigned int cmd_in, unsigned long arg)
 		result = get_user(val, ip);
 		if (result)
 			return result;
-		if (val > SG_MAX_CDB_SIZE)
-			return -ENOMEM;
 		sfp->next_cmd_len = (val > 0) ? val : 0;
 		return 0;
 	case SG_GET_VERSION_NUM:
