diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 3fb2aaa0..f9390a06 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -362,7 +362,7 @@ cmd_lzma = (cat $(filter-out FORCE,$^) | \
 
 quiet_cmd_lzo = LZO     $@
 cmd_lzo = (cat $(filter-out FORCE,$^) | \
-	lzop -9 && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
+	lzop -6 && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
 	(rm -f $@ ; false)
 
 quiet_cmd_lz4 = LZ4     $@
diff --git a/scripts/gen_initramfs_list.sh b/scripts/gen_initramfs_list.sh
index 17fa9014..39d6f3b0 100755
--- a/scripts/gen_initramfs_list.sh
+++ b/scripts/gen_initramfs_list.sh
@@ -254,7 +254,7 @@ case "$arg" in
                 && compr="xz --check=crc32 --lzma2=dict=1MiB"
 		echo "$output_file" | grep -q "\.lzo$" \
                 && [ -x "`which lzop 2> /dev/null`" ] \
-                && compr="lzop -9 -f"
+                && compr="lzop -6 -f"
 		echo "$output_file" | grep -q "\.lz4$" \
                 && [ -x "`which lz4 2> /dev/null`" ] \
                 && compr="lz4 -l -9 -f"
