From ae4fef6fc558b0621532d0de6ebaaeaabe861792 Mon Sep 17 00:00:00 2001
From: sub77 <sub77@ymail.com>
Date: Fri, 31 Jul 2015 17:47:45 +0200
Subject: [PATCH] fix build error: PS2MOUSE fsp_detect & lifebook

---
 drivers/input/mouse/psmouse-base.c | 62 +++++++++++++++++++-------------------
 1 file changed, 31 insertions(+), 31 deletions(-)

diff --git a/drivers/input/mouse/psmouse-base.c b/drivers/input/mouse/psmouse-base.c
index a19c183557d..ad3d2e861d0 100644
--- a/drivers/input/mouse/psmouse-base.c
+++ b/drivers/input/mouse/psmouse-base.c
@@ -708,14 +708,14 @@ static int psmouse_extensions(struct psmouse *psmouse,
 /*
  * We always check for lifebook because it does not disturb mouse
  * (it only checks DMI information).
+ *
+ *	if (psmouse_do_detect(lifebook_detect, psmouse, set_properties) == 0) {
+ *		if (max_proto > PSMOUSE_IMEX) {
+ *			if (!set_properties || lifebook_init(psmouse) == 0)
+ *				return PSMOUSE_LIFEBOOK;
+ *		}
+ *	}
  */
-	if (psmouse_do_detect(lifebook_detect, psmouse, set_properties) == 0) {
-		if (max_proto > PSMOUSE_IMEX) {
-			if (!set_properties || lifebook_init(psmouse) == 0)
-				return PSMOUSE_LIFEBOOK;
-		}
-	}
-
 /*
  * Try Kensington ThinkingMouse (we try first, because synaptics probe
  * upsets the thinkingmouse).
@@ -821,19 +821,19 @@ static int psmouse_extensions(struct psmouse *psmouse,
 /*
  * Try Finger Sensing Pad. We do it here because its probe upsets
  * Trackpoint devices (causing TP_READ_ID command to time out).
- */
-	if (max_proto > PSMOUSE_IMEX) {
-		if (psmouse_do_detect(fsp_detect,
-				      psmouse, set_properties) == 0) {
-			if (!set_properties || fsp_init(psmouse) == 0)
-				return PSMOUSE_FSP;
-/*
+ *
+ *	if (max_proto > PSMOUSE_IMEX) {
+ *		if (psmouse_do_detect(fsp_detect,
+ *				      psmouse, set_properties) == 0) {
+ *			if (!set_properties || fsp_init(psmouse) == 0)
+ *				return PSMOUSE_FSP;
+ *
  * Init failed, try basic relative protocols
+ *
+ *			max_proto = PSMOUSE_IMEX;
+ *		}
+ *	}
  */
-			max_proto = PSMOUSE_IMEX;
-		}
-	}
-
 /*
  * Reset to defaults in case the device got confused by extended
  * protocol probes. Note that we follow up with full reset because
@@ -944,12 +944,12 @@ static const struct psmouse_protocol psmouse_protocols[] = {
 	},
 #endif
 #ifdef CONFIG_MOUSE_PS2_LIFEBOOK
-	{
-		.type		= PSMOUSE_LIFEBOOK,
-		.name		= "LBPS/2",
-		.alias		= "lifebook",
-		.init		= lifebook_init,
-	},
+/*	{
+*		.type		= PSMOUSE_LIFEBOOK,
+*		.name		= "LBPS/2",
+*		.alias		= "lifebook",
+*		.init		= lifebook_init,
+*	},*/
 #endif
 #ifdef CONFIG_MOUSE_PS2_TRACKPOINT
 	{
@@ -985,13 +985,13 @@ static const struct psmouse_protocol psmouse_protocols[] = {
 	},
 #endif
 #ifdef CONFIG_MOUSE_PS2_SENTELIC
-	{
-		.type		= PSMOUSE_FSP,
-		.name		= "FSPPS/2",
-		.alias		= "fsp",
-		.detect		= fsp_detect,
-		.init		= fsp_init,
-	},
+/*	{
+*		.type		= PSMOUSE_FSP,
+*		.name		= "FSPPS/2",
+*		.alias		= "fsp",
+*		.detect		= fsp_detect,
+*		.init		= fsp_init,
+*	},*/
 #endif
 	{
 		.type		= PSMOUSE_CORTRON,
