diff --git a/decon-int_drv.c b/decon-int_drv.c
index 7695b67..b3010c9 100755
--- a/drivers/video/fbdev/exynos/decon_7570/decon-int_drv.c
+++ b/drivers/video/fbdev/exynos/decon_7570/decon-int_drv.c
@@ -440,11 +440,11 @@ int decon_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
 	case 24:
 		/* our 24bpp is unpacked, so 32bpp */
 		var->bits_per_pixel	= 32;
-		var->red.offset		= 16;
+		var->red.offset		= 0;
 		var->red.length		= 8;
 		var->green.offset	= 8;
 		var->green.length	= 8;
-		var->blue.offset	= 0;
+		var->blue.offset	= 16;
 		var->blue.length	= 8;
 		break;
 
