diff --git a/drivers/tspdrv/tspdrv.c b/drivers/tspdrv/tspdrv.c
index 69bdc46..8101fa9 100644
--- a/drivers/tspdrv/tspdrv.c
+++ b/drivers/tspdrv/tspdrv.c
@@ -92,11 +92,11 @@ static int g_nMajor = 0;
 
 
 /* Needs to be included after the global variables because they use them */
-#include <tspdrvOutputDataHandler.c>
+#include "tspdrvOutputDataHandler.c"
 #ifdef CONFIG_HIGH_RES_TIMERS
-    #include <VibeOSKernelLinuxHRTime.c>
+    #include "VibeOSKernelLinuxHRTime.c"
 #else
-    #include <VibeOSKernelLinuxTime.c>
+    #include "VibeOSKernelLinuxTime.c"
 #endif
 
 asmlinkage void _DbgOut(int level, const char *fmt,...)
