diff --git a/drivers/gpu/drm/msm/sde_hdcp_2x.c b/drivers/gpu/drm/msm/sde_hdcp_2x.c
index 01e5e2c9e516..10aca3e30049 100644
--- a/drivers/gpu/drm/msm/sde_hdcp_2x.c
+++ b/drivers/gpu/drm/msm/sde_hdcp_2x.c
@@ -418,16 +418,14 @@ static void sde_hdcp_2x_cleanup_work(struct kthread_work *work)
 static u8 sde_hdcp_2x_stream_type(u8 min_enc_level)
 {
 	u8 stream_type = 0;
-	u8 const hdcp_min_enc_level_0 = 0, hdcp_min_enc_level_1 = 1,
-	   hdcp_min_enc_level_2 = 2;
 	u8 const stream_type_0 = 0, stream_type_1 = 1;
 
 	switch (min_enc_level) {
-	case hdcp_min_enc_level_0:
-	case hdcp_min_enc_level_1:
+	case 0:
+	case 1:
 		stream_type = stream_type_0;
 		break;
-	case hdcp_min_enc_level_2:
+	case 2:
 		stream_type = stream_type_1;
 		break;
 	default:
