Defines PLATFORM_VERSION, which is needed for compillation.

diff --git a/Makefile b/Makefile
index 45ce83161..fdb71c8ec 100755
--- a/Makefile
+++ b/Makefile
@@ -469,6 +469,7 @@ asm-generic:
 	$(Q)$(MAKE) -f $(srctree)/scripts/Makefile.asm-generic \
 	            src=uapi/asm obj=arch/$(SRCARCH)/include/generated/uapi/asm
 
+PLATFORM_VERSION=9
 ifneq ($(PLATFORM_VERSION), )
 PLATFORM_VERSION_NUMBER=$(shell $(CONFIG_SHELL) $(srctree)/scripts/android-version.sh $(PLATFORM_VERSION))
 MAJOR_VERSION=$(shell $(CONFIG_SHELL) $(srctree)/scripts/android-major-version.sh $(PLATFORM_VERSION))
