diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index b40e5154..a17af78d 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -158,7 +158,8 @@ ld_flags       = $(LDFLAGS) $(ldflags-y)
 dtc_cpp_flags  = -Wp,-MD,$(depfile).pre.tmp -nostdinc                    \
 		 -I$(srctree)/arch/$(SRCARCH)/boot/dts                   \
 		 -Iarch/$(SRCARCH)/boot/dts				 \
-		 -I$(srctree)/arch/$(SRCARCH)/boot/dts/include           \
+                 -I$(srctree)/arch/$(SRCARCH)/boot/dts/include           \
+                 -I$(srctree)/include/                                   \
 		 -I$(objtree)/include/                                   \
 		 -I$(srctree)/drivers/of/testcase-data                   \
 		 -I$(objtree)/include					 \
