diff --git a/fs/ext4/super.c b/fs/ext4/super.c
index 414e8b11..d66fd851 100644
--- a/fs/ext4/super.c
+++ b/fs/ext4/super.c
@@ -4150,8 +4150,8 @@ cantfind_ext4:
 	/* for debugging, sangwoo2.lee */
 	/* If you wanna use the flag 'MS_SILENT', call */
 	/* 'print_bh' function within below 'if'. */
-	printk(KERN_ERR "printing data of superblock-bh\n");
-	print_bh(sb, bh, 0, EXT4_BLOCK_SIZE(sb));
+	//printk(KERN_ERR "printing data of superblock-bh\n");
+	//print_bh(sb, bh, 0, EXT4_BLOCK_SIZE(sb));
 	/* for debugging */
 
 	if (!silent)
