diff --git a/arch/arm/mach-msm/smd_init_dt.c b/arch/arm/mach-msm/smd_init_dt.c
index 640656cb..046f4470 100644
--- a/arch/arm/mach-msm/smd_init_dt.c
+++ b/arch/arm/mach-msm/smd_init_dt.c
@@ -21,7 +21,7 @@
 #include <linux/io.h>
 
 #include <mach/msm_ipc_logging.h>
-#include <smd_private.h>
+#include "smd_private.h"
 
 #define MODULE_NAME "msm_smd"
 #define IPC_LOG(level, x...) do { \
diff --git a/sound/soc/codecs/msm8x10-wcd-tables.c b/sound/soc/codecs/msm8x10-wcd-tables.c
index 7e0263d4..9167dd83 100644
--- a/sound/soc/codecs/msm8x10-wcd-tables.c
+++ b/sound/soc/codecs/msm8x10-wcd-tables.c
@@ -10,7 +10,7 @@
  * GNU General Public License for more details.
  */
 
-#include <msm8x10_wcd_registers.h>
+#include "msm8x10_wcd_registers.h"
 #include "msm8x10-wcd.h"
 
 const u8 msm8x10_wcd_reg_readable[MSM8X10_WCD_CACHE_SIZE] = {
diff --git a/sound/soc/msm/msm8x10.c b/sound/soc/msm/msm8x10.c
index 5e666f0f..e69d0fef 100644
--- a/sound/soc/msm/msm8x10.c
+++ b/sound/soc/msm/msm8x10.c
@@ -26,7 +26,7 @@
 #include <sound/jack.h>
 #include <asm/mach-types.h>
 #include <mach/socinfo.h>
-#include <qdsp6v2/msm-pcm-routing-v2.h>
+#include "../qdsp6v2/msm-pcm-routing-v2.h"
 #include <sound/q6afe-v2.h>
 #include <linux/module.h>
 #include <mach/gpiomux.h>
diff --git a/sound/soc/msm/qdsp6v2/rtac.c b/sound/soc/msm/qdsp6v2/rtac.c
index 701dfef6..8d2e1d41 100644
--- a/sound/soc/msm/qdsp6v2/rtac.c
+++ b/sound/soc/msm/qdsp6v2/rtac.c
@@ -25,7 +25,7 @@
 #include <sound/q6asm-v2.h>
 #include <sound/q6afe-v2.h>
 #include <sound/apr_audio-v2.h>
-#include <q6voice.h>
+#include "q6voice.h"
 #include "audio_acdb.h"


diff --git a/sound/soc/msm/msm8226.c b/sound/soc/msm/msm8226.c
index 7ff4c82..70322dd 100644
--- a/sound/soc/msm/msm8226.c
+++ b/sound/soc/msm/msm8226.c
@@ -27,7 +27,7 @@
 #include <asm/mach-types.h>
 #include <mach/socinfo.h>
 #include <mach/subsystem_notif.h>
-#include <qdsp6v2/msm-pcm-routing-v2.h>
+#include "qdsp6v2/msm-pcm-routing-v2.h"
 #include "qdsp6v2/q6core.h"
 #include "../codecs/wcd9xxx-common.h"
 #include "../codecs/wcd9306.h" 
 
